### PR TITLE
Issue #6320: resolve remove conditional for EqualsAvoidNullCheck

### PR DESCRIPTION
**Pull request by rnveach**
Issue #6320
http://rveach.no-ip.org/checkstyle/regression/pitest-reports/37/pitest-coding/com.puppycrawl.tools.checkstyle.checks.coding/EqualsAvoidNullCheck.java.html

Regression to come.

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java [hunk 1]
```diff
@@ -288,9 +287,8 @@ private void leaveLiteralNew(DetailAST ast) {
      */
     private void traverseFieldFrameTree(FieldFrame frame) {
         for (FieldFrame child: frame.getChildren()) {
-            if (!child.getChildren().isEmpty()) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java]: This just prevents calling the same method when we know the iterator in the new method call will be empty and won't do anything.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java [hunk 2]
```diff
@@ -573,9 +570,7 @@ public void addChild(FieldFrame child) {
          * @param field the ast of the field.
          */
         public void addField(DetailAST field) {
-            if (field.findFirstToken(TokenTypes.IDENT) != null) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java]: This was resolved during the write up. Re-pushing.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java [hunk 3]
```diff
@@ -233,8 +233,7 @@ private void processFrame(DetailAST ast) {
         final FieldFrame frame = new FieldFrame(currentFrame);
         final int astType = ast.getType();
         if (astType == TokenTypes.CLASS_DEF
-                || astType == TokenTypes.ENUM_DEF
-                || astType == TokenTypes.ENUM_CONSTANT_DEF) {
+                || astType == TokenTypes.ENUM_DEF) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java]: Unsure how to kill this.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java [hunk 4]
```diff
@@ -363,8 +361,7 @@ private static boolean containsAllSafeTokens(final DetailAST expr) {
      */
     private static DetailAST skipVariableAssign(final DetailAST currentAST) {
         DetailAST result = currentAST;
-        if (currentAST.getType() == TokenTypes.ASSIGN
-                && currentAST.getFirstChild().getType() == TokenTypes.IDENT) {
+        if (currentAST.getType() == TokenTypes.ASSIGN) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java]: Unsure how to kill this.

**Comment by rnveach** [comment]
Regression: http://rveach.no-ip.org/checkstyle/regression/reports/268/

No differences.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java (hunk 4 above)
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java]: An `ASSIGN` expressions requires a variable to store the value. So the first children will be an `IDENT` for sure.

**Comment by romani** [comment]
Conflict, please resolve merge

**Comment by rnveach** [comment]
Rebased.

**Pull request merged**